x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtAuthenticator.java

```diff
@@ -46,9 +48,16 @@ public JwtAuthenticator(
         this.realmConfig = realmConfig;
         this.tokenType = realmConfig.getSetting(JwtRealmSettings.TOKEN_TYPE);
         if (tokenType == JwtRealmSettings.TokenType.ID_TOKEN) {
+            this.fallbackClaimNames = Map.of();
             this.jwtFieldValidators = configureFieldValidatorsForIdToken(realmConfig);
         } else {
-            this.jwtFieldValidators = configureFieldValidatorsForAccessToken(realmConfig);
+            this.fallbackClaimNames = Map.of(
+                "sub",
+                realmConfig.getSetting(JwtRealmSettings.FALLBACK_SUB_CLAIM),
+                "aud",
+                realmConfig.getSetting(JwtRealmSettings.FALLBACK_AUD_CLAIM)
+            );
```

Comment:
Because the `K` and `V` types of the Map are the same, it feels like this would be better to use `Map.ofEntries`.

It's easy to get an argument (or 2) wrong here, and end up with the wrong Map (if `K` and `V` are different that can't happen so easily).

Reply:
This is a good point. I changed it to use `Map.ofEntries`.